client-app/src/examples/contact/mobile/AppModel.ts

```diff
@@ -0,0 +1,98 @@
+import {managed, PlainObject, XH} from '@xh/hoist/core';
+import {ContactService} from '../svc/ContactService';
+import {BaseAppModel} from '../../../BaseAppModel';
+import {NavigatorModel} from '@xh/hoist/mobile/cmp/navigator';
+import directoryPanel from './DirectoryPanel';
+import detailsPanel from './details/DetailsPanel';
+import DetailsPanelModel from './details/DetailsPanelModel';
+import DirectoryPanelModel from './DirectoryPanelModel';
+import {action, makeObservable, observable, runInAction} from '@xh/hoist/mobx';
+import {castArray, uniq} from 'lodash';
+
+import {
+    autoRefreshAppOption,
+    sizingModeAppOption,
+    themeAppOption
+} from '@xh/hoist/mobile/cmp/appOption';
+
+export default class AppModel extends BaseAppModel {
+    static instance: AppModel;
+
+    @observable.ref tagList: string[] = [];
+    @observable.ref contacts: PlainObject[] = [];
+
+    @managed directoryPanelModel: DirectoryPanelModel;
+    @managed detailsPanelModel: DetailsPanelModel;
+
+    @managed
+    navigatorModel: NavigatorModel = new NavigatorModel({
+        track: true,
+        pages: [
+            {id: 'default', content: directoryPanel},
+            {id: 'details', content: detailsPanel}
+        ]
+    });
+
+    override getRoutes() {
+        return [
+            {
+                name: 'default',
+                path: '/contactMobile',
+                children: [
+                    {
+                        name: 'details',
+                        path: '/:id'
+                    }
+                ]
+            }
+        ];
+    }
+
+    constructor() {
+        super();
+        makeObservable(this);
+        this.directoryPanelModel = new DirectoryPanelModel(this);
+        this.detailsPanelModel = new DetailsPanelModel(this);
+    }
+
+    async updateContactAsync(id: string, data: PlainObject = {}) {
```

Comment:
Is `data` optional here? I don't see why you call this without `data`?

If not, I'd recommend we lose the default `{}` value, and instead return early if `data` is null, to avoid the call for to the server for a no-op.

Reply:
Done.